### DISCUSSION TITLE
Query Subcollection

### DISCUSSION THREAD

**Discussion by Rol3oT** [GodotNuts/GodotFirebase#452] [Q&A]
Hello,

I have a Firebase Database structure as follows:
users(collection)/user_id(documents)/pets(collection)/pets_id(document)

Trying to get query working for collecting all pets documents of a specific user. I got it working on the firebase query builder
![image](https://github.com/user-attachments/assets/2fc0815c-2c5e-46d9-8d23-c627d4b8f597)

and I can get query working for the first layer of collections users:
```
var users_collection_path = "users"
var pets_query : FirestoreQuery = FirestoreQuery.new()
pets_query.from(users_collection_path)
pets_query.limit(100)
var pets_data = await Firebase.Firestore.query(pets_query)
print(pets_data)
```
![image](https://github.com/user-attachments/assets/9eca9c9f-1350-4349-9a63-30fa123009be)

**but I do not get any subcollections even thou `all_descendants `is true(by default)**

I also tried building query for subcollections directly as in query builder, but get an error
```
var pets_subcollection_path = USER_COLLECTION_ID + "/" + auth.localid + "/" + PET_COLLECTION_ID
var pets_query : FirestoreQuery = FirestoreQuery.new()
pets_query.from(pets_subcollection_path)
pets_query.limit(100)
var pets_data = await Firebase.Firestore.query(pets_query)
print(pets_data)
```
but get "Nil" error
![image](https://github.com/user-attachments/assets/dbe6bfd9-2a07-4b3f-876c-da2fa3cc85aa)

Am I doing something wrong here? Any help would be greatly appreciated! If this doesn't work I could restructure my database with not having any subcollections ever, but I think there should be a way to get this working

**Comment by Rol3oT** [comment]
Also tried creating an exception in the Firestore Database, but I don't think this is it, since I am looking for a subcollection
![image](https://github.com/user-attachments/assets/337e0079-114c-445f-a92a-f496845ab227)

**Reply by WolfgangSenff** [reply]
It is it. You need to enable it for the collection group, but you haven't. You can see it to the right.

**Reply by Rol3oT** [reply]
Doesn't seem like you can change the automatic index settings ones if thats what you mean.
Tried to create a wildcard field path one I found mentioned on stack overflow, but still only gives me top level documents and not subcollections
![image](https://github.com/user-attachments/assets/7a873749-6b83-4e7d-9bea-016fe3abc916)

![image](https://github.com/user-attachments/assets/e1e78bf0-9555-4491-a6b0-721f2607ed5d)

**Reply by WolfgangSenff** [reply]
It looks like you might have it setup right, although it's hard to tell. I was mistaken based on not having a pic of my own up when I was speaking earlier. 
![Screenshot_20241228-232232](https://github.com/user-attachments/assets/fa1281b2-b363-4b75-b5f6-c9b615b55da7)

Well, the only thing I can tell is it works for others when we do this, myself included, but not you. My only real recommendation is to figure out either how to run the query with a Cloud Function, or restructure your database so you're not going against sub collections like that.
